examples-gallery/plot_Mississippi_steamboat.py

```diff
@@ -0,0 +1,445 @@
+# %%
+"""
+Mississippi Steamboat
+=====================
+
+A boat is modeled as a rectangular plate with length :math:`a_S` and
+width :math:`b_S`.
+It has a mass :math:`m_S` and is modeled as a rigid body.
+Water wheels are attached to the boat on the left and right side.
+The wheels have radius :math:`r_W` and mass :math:`m_W` and are modeled as
+rigid bodies.
+By running the wheels at different speeds, the boat can be steered.
+The water speed is assumed to be zero. Gravity, in the negative Z direction, is
+unimportant here, hence disregarded.
+
+**Constants**
+
+- :math:`m_S`: mass of the steamboat [kg]
+- :math:`m_W`: mass of the wheels [kg]
+- :math:`r_W`: radius of the wheel [m]
+- :math:`a_S`: length of the steamboat [m]
+- :math:`b_S`: width of the steamboat [m]
+- :math:`c_W`: drag coefficient at wheels [kg/m^2]
+- :math:`c_S`: drag coefficient at steamboat [kg/m^2]
+
+
+**States**
+
+- :math:`x`: X - position of the center of the steamboat [m]
+- :math:`y`: Y - position of the center of the steamboat [m]
+- :math:`q_S`: angle of the steamboat [rad]
+- :math:`q_{LW}`: angle of the left wheel [rad]
+- :math:`q_{RW}`: angle of the right wheel [rad]
+- :math:`u_x`: speed of the steamboat in X direction [m/s]
+- :math:`u_y`: speed of the steamboat in Y direction [m/s]
+- :math:`u_S`: angular speed of the steamboat [rad/s]
+- :math:`u_{LW}`: angular speed of the left wheel [rad/s]
+- :math:`u_{RW}`: angular speed of the right wheel [rad/s]
+
+**Specifieds**
+
+- :math:`t_{LW}`: torque applied to the left wheel [Nm]
+- :math:`t_{RW}`: torque applied to the right wheel [Nm]
+
+"""
+import numpy as np
+import sympy as sm
+from opty.utils import parse_free
+from scipy.interpolate import CubicSpline
+import sympy.physics.mechanics as me
+from opty.direct_collocation import Problem
+import matplotlib.pyplot as plt
+from matplotlib.animation import FuncAnimation
+from matplotlib.patches import Rectangle
+
+
+# %%
+# Set up the Equations of Motion.
+# -------------------------------
+#
+# Set up the geometry of the system.
+#
+# - :math:`N`: inertial frame of reference
+# - :math:`O`: origin of the inertial frame of reference
+# - :math:`A_S`: body fixed frame of the steamboat
+# - :math:`A_{LW}`: body fixed frame of the left wheel
+# - :math:`A_{RW}`: body fixed frame of the right wheel
+# - :math:`Dmc_S`: center of the steamboat
+# - :math:`Dmc_{LW}`: center of the left wheel
+# - :math:`Dmc_{RW}`: center of the right wheel
+# - :math:`FP_{LW}`: point where the force due to rotation of the left
+#   wheel is applied
+# - :math:`FP_{RW}`: point where the force due to rotation of the right
+#   wheel is applied
+
+N, AS, ALW, ARW = sm.symbols('N, AS, ALW, ARW', cls = me.ReferenceFrame)
```

Comment:
```suggestion
N, AS, ALW, ARW = sm.symbols('N, AS, ALW, ARW', cls=me.ReferenceFrame)
```

PEP8: no space around kwarg equals signs.